### PR TITLE
variable, ddl: add session variable 'tidb_ddl_reorg_priority' to set operation priority of ddl reorg

### PR DESCRIPTION
**Pull request by winkyao**
## What have you changed? (mandatory)

Add session variable 'tidb_ddl_reorg_priority' to set operation priority of ddl reorg. Before this PR, the operation in adding indices is definitely in low priority, In some case, we want to finish DDL job as fast as possible, after this PR, we can set the variable `tidb_ddl_reorg_priority` to `priority_high` to achieve it. 

## What is the type of the changes? (mandatory)

- New feature (non-breaking change which adds functionality)

## How has this PR been tested? (mandatory)

UT

## Does this PR affect documentation (docs/docs-cn) update? (mandatory)

YES

## Does this PR affect tidb-ansible update? (mandatory)

NO

## Does this PR need to be added to the release notes? (mandatory)

YES:

```
add session variable 'tidb_ddl_reorg_priority' to set operation priority of DDL reorg.
```

### CODE REVIEW

**Comment by shenli** [comment]
Please add proper labels.

**Comment by shenli** [comment]
This PR should be metioned in the release notes.

**Comment by winkyao** [comment]
@ciscoxll @zimulala @crazycs520 PTAL

**Comment by ciscoxll** [comment]
@winkyao Please resolve conflicts.

**Comment by winkyao** [comment]
@ciscoxll @zimulala @crazycs520 PTAL

File: executor/executor_test.go [hunk 1]
```diff
@@ -2211,6 +2211,34 @@ func (s *testContextOptionSuite) TestAddIndexPriority(c *C) {
...
+	cli.mu.Lock()
+	cli.mu.checkFlags = checkDDLAddIndexPriority
+	cli.mu.Unlock()
+
+	cli.priority = pb.CommandPri_High
+	tk.MustExec("alter table t1 add index t1_index (id);")
+
+	cli.mu.Lock()
+	cli.mu.checkFlags = checkRequestOff
+	cli.mu.Unlock()
```
Comment (crazycs520) [inline executor/executor_test.go]: I can't understand the test is used to test what?

Comment (winkyao) [reply executor/executor_test.go]: test if following set priority operation takes effect.

Comment (crazycs520) [reply executor/executor_test.go]: Which line to check if the following set priority operation takes effect?

Comment (winkyao) [reply executor/executor_test.go]: You can look into checkRequestClient.SendRequest

Comment (crazycs520) [reply executor/executor_test.go]: Got it~

File: meta/meta.go [hunk 2]
```diff
@@ -504,8 +504,16 @@ func (m *Meta) getDDLJob(key []byte, index int64) (*model.Job, error) {
 		return nil, errors.Trace(err)
 	}
 
-	job := &model.Job{}
+	job := &model.Job{
+		// for compability, if the job is enqueued by old version TiDB and Priority field is omitted,
+		// set the default priority to kv.PriorityLow
```
Comment (crazycs520) [inline meta/meta.go]: end with .

File: meta/meta.go [hunk 3]
```diff
@@ -504,8 +504,16 @@ func (m *Meta) getDDLJob(key []byte, index int64) (*model.Job, error) {
 		return nil, errors.Trace(err)
 	}
 
-	job := &model.Job{}
+	job := &model.Job{
+		// for compability, if the job is enqueued by old version TiDB and Priority field is omitted,
```
Comment (crazycs520) [inline meta/meta.go]: s/for/For

File: meta/meta.go [hunk 4]
```diff
@@ -504,8 +504,16 @@ func (m *Meta) getDDLJob(key []byte, index int64) (*model.Job, error) {
...
+		// set the default priority to kv.PriorityLow
+		Priority: kv.PriorityLow,
+	}
 	err = job.Decode(value)
+	// check job.Priority is valid.
```
Comment (crazycs520) [inline meta/meta.go]: s/check/Check

File: sessionctx/variable/session.go [hunk 5]
```diff
@@ -453,6 +458,18 @@ func (s *SessionVars) deleteSystemVar(name string) error {
...
 
+func (s *SessionVars) setDDLReorgPriority(val string) {
+	val = strings.ToLower(val)
+	switch val {
+	case "priority_low":
+		s.DDLReorgPriority = kv.PriorityLow
+	case "priority_normal":
+		s.DDLReorgPriority = kv.PriorityNormal
+	case "priority_high":
+		s.DDLReorgPriority = kv.PriorityHigh
```
Comment (zhexuany) [inline sessionctx/variable/session.go]: maybe add a default clause?

Comment (ciscoxll) [reply sessionctx/variable/session.go]: Do agree with @zhexuany.

**Comment by ciscoxll** [comment]
reset LGTM.

File: sessionctx/variable/session.go (hunk 5 above)
Comment (winkyao) [reply sessionctx/variable/session.go]: Done

**Comment by ciscoxll** [comment]
LGTM

**Pull request merged**